internal/databases/postgres/query_runner.go

```diff
@@ -238,6 +238,16 @@ func (queryRunner *PgQueryRunner) StopBackup() (label string, offsetMap string,
 	defer queryRunner.Mu.Unlock()
 
 	tracelog.InfoLogger.Println("Calling pg_stop_backup()")
+
+	//during long backups connection might break, so we check if connection is still alive
+	errPing := queryRunner.Connection.Ping(context.TODO())
+	if errPing != nil {
+		queryRunner.Connection, err = Connect()
+		if err != nil {
+			return "", "", "", fmt.Errorf("QueryRunner StopBackup: connection is dead: %w %w", errPing, err)
+		}
+	}
+
```

Comment:
JFYI: this will not work for Cloudberry...

```
* If the client disconnects with a non-exclusive backup running, the backup
is automatically aborted. This is the same thing that pg_basebackup does.
To use these non-exclusive backups the backup software will have to
maintain a persistent connection to the database -- something that should
not be a problem for any of the modern ones out there (but would've been
slightly trickier back in the days when we suggested shellscripts)
```